OpenEdXMobile/src/main/java/org/edx/mobile/base/MainApplication.java

```diff
@@ -132,16 +130,32 @@ private void init() {
         );
     }
 
-    private boolean needVersionUpgrade(Context context) {
-        boolean needVersionUpgrade = false;
-        PrefManager.AppInfoPrefManager pmanager = new PrefManager.AppInfoPrefManager(context);
-        Long previousVersion = pmanager.getAppVersionCode();
-        final int curVersion = BuildConfig.VERSION_CODE;
-        if (previousVersion < curVersion) {
-            needVersionUpgrade = true;
-            pmanager.setAppVersionCode(curVersion);
+    private void checkIfAppVersionUpgraded(Context context) {
+        PrefManager.AppInfoPrefManager prefManager = new PrefManager.AppInfoPrefManager(context);
+        long previousVersionCode = prefManager.getAppVersionCode();
+        final long curVersionCode = (long) BuildConfig.VERSION_CODE;
+        if (previousVersionCode < 0) {
```

Comment:
@BenjiLee Yes it will be i-e- "-1" but kept condition more generic to consider all less than 0 as first opening (as this default -1 could be changed).
Equal comparison with -1 will be good too at this time.